rmc-docs/src/tutorial-loops-unwinding.md

```diff
@@ -0,0 +1,129 @@
+# Loops, unwinding, and bounds
+
+Consider code like this:
+
+```rust
+{{#include tutorial/loops-unwinding/src/lib.rs:code}}
+```
+
+And a proof harness like this:
+
+```rust
+{{#include tutorial/loops-unwinding/src/lib.rs:rmc}}
+```
+
+When we run RMC on this, we run into an unfortunate result: non-termination.
+
+> **NOTE:** Presently, [due to a bug](https://github.com/model-checking/rmc/issues/493), this is especially bad: we don't see any output at all.
+> You are supposed to see some log lines that might give some clue that an infinite loop is occurring.
+> If RMC doesn't terminate, it's almost always the problem that this section is covering, however.
+
+To verify programs like this, we really need to do two things:
+
+1. Create an upper bound on the size of the problem.
+We've actually already done part of this: our proof harness seems to be trying to set an upper limit of 10.
+2. Tell RMC about this limit, if it's not able to figure it out on its own.
+
+> **NOTE:** In the future, RMC may eventually support specifying _loop invariants_, which allow us to do away with fixed upper bounds like this.
+> That support is not ready yet, however.
+
+Bounding proofs like this means we may no longer be proving as much as we originally hoped.
+Who's to say, if we prove everything works up to size 10, there isn't a novel bug lurking, expressible only with problems of size 11+?
+But, let's get back to the practical issue at hand.
+
+We can "make progress" in our work by giving RMC a global bound on the problem size.
+RMC will automatically generate verification conditions that help us understand if that bound isn't enough.
+Let's start with the "sledge hammer" by dropping all the way down to size 1:
+
+```
+# rmc src/lib.rs --cbmc-args --unwind 1
+[.unwind.0] unwinding assertion loop 0: FAILURE
+VERIFICATION FAILED
+```
+
+This output is showing us two things:
+
+1. RMC tells us we haven't unwound enough. This is the failure of the "unwinding assertion."
+2. We aren't seeing other failures if we only unroll the loop once.
+We've basically turned the loop into an `if`, and that's not enough to show the bug.
```

Comment:
> and that's not enough to show the bug.
There's no preceding explanation of what the bug in that code is, and it might not be obvious to everyone, so you might want to add a sentence at the beginning of the section (after the code snippet) that clarifies that this code has a bug due to an out-of-bound access.